packages/redis_client/lib/src/redis_client.dart

```diff
@@ -73,153 +141,169 @@ class RedisClient {
   /// A completer which completes when the client establishes a connection.
   var _connected = Completer<void>();
 
+  /// Whether the client is connected.
+  var _isConnected = false;
+
   /// A completer which completes when the client disconnects.
   /// Begins in a completed state since the client is initially disconnected.
   var _disconnected = Completer<void>()..complete();
 
   /// A future which completes when the client establishes a connection.
-  Future<void> get connected => _connected.future;
+  Future<void> get _untilConnected => _connected.future;
```

Comment:
This naming raised an eyebrow but seeing it in use is 💯

Reply:
haha thanks glad you thought so!